src/enfa.ts

```diff
@@ -168,9 +203,138 @@ class EpsilonNFABuilder {
       case 'NamedBackRef': {
         throw new Error('unimplemented');
       }
+      case 'LineBegin':
+      case 'LineEnd': {
+        const q0 = this.throughState();
+        return {
+          initialState: q0,
+          acceptingState: q0,
+        };
+      }
     }
   }
 
+  private buildSubMatch(
+    pattern: Node,
+    initialState: State,
+    acceptingState: State,
+  ): [initialState: State, acceptingState: State] {
+    if (pattern.type === 'Disjunction') {
+      // 始端submatch (非貪欲)
+      const oldInitialTransitions = this.transitions.filter(
+        (t) => t[0] === initialState,
+      );
+      oldInitialTransitions.forEach(([source, node, dest], index) => {
+        const patChild: Node = pattern.children[index];
+        if (
+          (patChild.type === 'Sequence' &&
+            patChild.children.length > 0 &&
+            patChild.children[0].type !== 'LineBegin') ||
+          (patChild.type === 'Sequence' && patChild.children.length === 0) ||
+          (patChild.type !== 'Sequence' && patChild.type !== 'LineBegin')
+        ) {
+          // 元の遷移を削除
+          this.transitions = this.transitions.filter(
+            ([s, n, d]) => s !== source || n !== node || d !== dest,
```

Comment:
```suggestion
            ([s, n, d]) => !(s === source && n === node && d === dest),
```
これ初期状態からpatChildへの遷移を消すという意味だと思うのですが意図が伝わりやすいようにこっちの方が良さそうです(完全に僕の好みですが)

Reply:
確かに、そのほうが可読性が高いと思いました。ありがとうございます